packages/runtime/src/runtime/websocket.tsx

```diff
@@ -0,0 +1,68 @@
+import {
+  type useScreenData,
+  type WebSocketConnection,
+  type EnsembleActionHookResult,
+} from "@ensembleui/react-framework";
+
+export const handleConnectSocket = (
+  screenData: ReturnType<typeof useScreenData>,
+  socketName: string,
+  onOpen?: EnsembleActionHookResult,
+  onMessage?: EnsembleActionHookResult,
+  onClose?: EnsembleActionHookResult,
+): WebSocketConnection | undefined => {
+  const socket = screenData.sockets?.find((model) => model.name === socketName);
+  if (!socket) {
+    return;
+  }
+
+  const ws = new WebSocket(socket.uri);
```

Comment:
You should check if there's an existing socket that is connected before opening a new one.